tests/testthat/test-format_data.R

```diff
@@ -1,442 +1,162 @@
-test_that("format_pkncaconc_data generates correct dataset", {
-  # Generate sample ADNCA data
-  ADNCA <- data.frame(
-    STUDYID = rep(1, 10),
-    USUBJID = rep(1:2, each = 5),
-    PCSPEC = rep("Plasma", 10),
-    DRUG = rep("DrugA", 10),
-    PARAM = rep("Analyte1", 10),
-    DOSNO = rep(1, 10),
-    AFRLT = seq(0, 9),
-    ARRLT = seq(0, 9),
-    AVAL = runif(10),
-    ROUTE = c("intravascular")
-  )
-
-  # Call format_pkncaconc_data
-  df_conc <- format_pkncaconc_data(ADNCA,
-                                   group_columns = c("STUDYID", "USUBJID", "PCSPEC",
-                                                     "DRUG", "PARAM"),
-                                   time_column = "AFRLT",
-                                   rrlt_column = "ARRLT")
-
-  # Test if df_conc is a data frame
-  expect_s3_class(df_conc, "data.frame")
-
-  # Test if df_conc has the correct columns
-  expect_true(all(c("STUDYID", "USUBJID", "PCSPEC", "DRUG", "PARAM",
-                    "AFRLT", "AVAL", "conc_groups", "TIME", "TIME_DOSE") %in% colnames(df_conc)))
-
-  # Test if df_conc is correctly grouped and arranged
-  expect_equal(df_conc$TIME, df_conc$AFRLT)
-
-  # Test if df_conc can be used with PKNCAconc by testing its output
-  expect_no_error(
-    PKNCA::PKNCAconc(
-      df_conc,
-      formula = AVAL ~ TIME | STUDYID + PCSPEC + DRUG + USUBJID / PARAM,
-      exclude_half.life = "exclude_half.life",
-      time.nominal = "NFRLT"
+# Shared data setup for all tests
+ADNCA <- data.frame(
+  STUDYID = rep(1, 20),
+  USUBJID = rep(1:2, each = 10),
+  PCSPEC = rep("Plasma", 20),
+  DRUG = rep("DrugA", 20),
+  PARAM = rep("Analyte1", 20),
+  AFRLT = rep(seq(0, 9), 2),
+  ARRLT = rep(seq(0, 4), 4),
+  NFRLT = rep(seq(0, 9), 2),
+  DOSNO = rep(1, 20),
+  DOSEA = rep(c(5, 10), each = 10),
+  ROUTE = rep(c("intravascular", "extravascular"), each = 10),
+  ADOSEDUR = rep(c(0, 0), each = 10),
+  AVAL = runif(20)
+)
+
+describe("format_pkncaconc_data", {
+  it("generates a valid dataset with required columns", {
+    df_conc <- format_pkncaconc_data(ADNCA,
+                                     group_columns = c("STUDYID", "USUBJID", "PCSPEC",
+                                                       "DRUG", "PARAM"),
+                                     time_column = "AFRLT",
+                                     rrlt_column = "ARRLT")
+
+    expect_s3_class(df_conc, "data.frame")
+    expect_true(all(c("STUDYID", "USUBJID", "PCSPEC", "DRUG", "PARAM",
+                      "AFRLT", "AVAL", "conc_groups", "TIME", "TIME_DOSE") %in% colnames(df_conc)))
```

Comment:
nitpick:
```suggestion
    expect_setequal(c("STUDYID", "USUBJID", "PCSPEC", "DRUG", "PARAM",
                      "AFRLT", "AVAL", "conc_groups", "TIME", "TIME_DOSE"), colnames(df_conc))
```
unless we are skipping some columns that are not required?